python/fairlearn-core/fairlearn/postprocessing/_curve_plotting_utilities.py

```diff
@@ -3,15 +3,19 @@
 
 """Utilities for plotting curves."""
 
-import matplotlib.pyplot as plt
-import matplotlib.cm as cm
-import matplotlib.colors
-
-tab10_norm = matplotlib.colors.Normalize(vmin=0, vmax=7)
-tab10_scalarMap = cm.ScalarMappable(norm=tab10_norm, cmap='Dark2')
-debug_colors = [tab10_scalarMap.to_rgba(x) for x in range(10)]
-debug_ncolors = len(debug_colors)
-debug_colormap = {}
+try:
+    import matplotlib.pyplot as plt
+    import matplotlib.cm as cm
+    import matplotlib.colors
+
+    tab10_norm = matplotlib.colors.Normalize(vmin=0, vmax=7)
+    tab10_scalarMap = cm.ScalarMappable(norm=tab10_norm, cmap='Dark2')
+    debug_colors = [tab10_scalarMap.to_rgba(x) for x in range(10)]
+    debug_ncolors = len(debug_colors)
+    debug_colormap = {}
+except ImportError:
+    print("Please install fairlearn-core[postprocessingplot] to get the dependencies for postprocessing plots.")
```

Comment:
Should you set a flag which can be checked in all the routines which would have called matplotlib, and print this error message again?

Reply:
There is a flag called "plot" that is True if we're supposed to plot the ROC curves. I can extend the checks to not only check for this flag, but also add a check to see if matplotlib is installed.